src/components/HearingList.js

```diff
@@ -9,6 +14,47 @@ import LabelList from './LabelList';
 import Label from 'react-bootstrap/lib/Label';
 import LoadSpinner from './LoadSpinner';
 import getAttr from '../utils/getAttr';
+import HearingsSearch from './HearingsSearch';
+import toLower from 'lodash/toLower';
+import upperFirst from 'lodash/upperFirst';
+
+
+const HearingListTabs = () =>
+  <Nav className="hearing-list__tabs" bsStyle="tabs" style={{width: '100%'}} activeKey="1" pullRight={1}>
+    <NavItem eventKey="3" disabled style={{width: '70%'}}/>
+    <NavItem eventKey="1" href="/home">Lista</NavItem>
+    <NavItem eventKey="2" title="Item">Kartta</NavItem>
+  </Nav>;
+
+const HearingListFilters = ({labels, handleChangeFilter, handleSort}) =>
+  <div className="hearing-list__filter-bar">
+    <FormGroup controlId="formControlsSelect" className="hearing-list__filter-bar-filter">
+      <ControlLabel><FormattedMessage id="sort"/></ControlLabel>
+      <FormControl componentClass="select" placeholder="select" onChange={(event) => handleSort(event.target.value)}>
+        <option value="-created_at"><FormattedMessage id="newestFirst"/></option>
+        <option value="created_at"><FormattedMessage id="oldestFirst"/></option>
+        <option value="-close_at"><FormattedMessage id="lastClosing"/></option>
+        <option value="close_at"><FormattedMessage id="firstClosing"/></option>
+        <option value="-open_at"><FormattedMessage id="lastOpen"/></option>
+        <option value="open_at"><FormattedMessage id="firstOpen"/></option>
+        <option value="-n_comments"><FormattedMessage id="mostCommented"/></option>
+        <option value="n_comments"><FormattedMessage id="leastCommented"/></option>
+      </FormControl>
+    </FormGroup>
+    {labels.length !== 0 && <FormGroup controlId="formControlsSelect" className="hearing-list__filter-bar-filter">
+      <ControlLabel><FormattedMessage id="chooseTopic"/></ControlLabel>
+      <FormControl componentClass="select" placeholder="select" onChange={(event) => handleChangeFilter(event.target.value)}>
```

Comment:
Hm, essentially text search and label search are both "filters" and as such could be treated equally. However, it might very well be that the user prefers to search by text (and not label), so the text search box could have primacy and the label could be shown as an additional filter. To me, though, it is more confusing have the label filter (which behaves like the text filter) next to the order control (which does not filter at all).

Any way, the location of the filter control shouldn't really affect the behavior of the filter:

1. a) In any case, the behavior should be AND (if the user has specified both text and Label filters), obtained with the backend `v1/hearing/?title=abc&label=1,3` query. Obviously, in most cases the user wouldn't fill both boxes simultaneously.
b) onChange would be preferable. However, it might be that the contains filter for the text in the backend is too slow, so onSubmit can be used if it feels more responsive.

2. a) The label autocomplete should (hopefully) work fast enough for onChange filtering of labels based on typing, so yes :)
b) onChange would be preferable but, again, if it's not responsive enough we'll have to try onSubmit instead.
c) I would suggest a multiple select field like https://github.com/JedWatson/react-select with `multi={true}`, or something similar.
d) In the backend multiple labels currently work with OR logic. Admittedly, it is a bit weird that a single filter works with OR logic and multiple filters combine with AND logic, but actually that seems to be our standard in endpoints which have multiple possible filters, so let's do that for now and see how it feels in practice.